### DISCUSSION TITLE
Remote rendering error

### DISCUSSION THREAD

**Discussion by hewittzgh** [Kitware/trame#75] [Q&A]
Hi everyone, when I execute the linked example (https://github.com/Kitware/trame-tutorial/blob/master/01_vtk/solution_ray_cast.py), I get the following error:

-------------------------------------------------------------------------
(The following is the stdout)

$python solution_ray_cast.py --port 23081 --server --host 0.0.0.0

App running at:
 - Local:   http://0.0.0.0:23081/
 - Network: http://127.0.1.1:23081/

Note that for multi-users you need to use and configure a launcher.
Segmentation fault (core dumped)

-------------------------------------------------------------------------
And after I change to local rendering mode, it works fine.
Can anyone tell me the reason for this error and the solution? Thank you!

**Comment by jourdain** [comment]
I'm guessing because you are running on a machine that does not have a graphical environment (X) setup and running?
For headless server, you will need an EGL or OSMesa version of VTK which for now, you can only get from a ParaView binary.

**Reply by tayran-mila** [reply]
Hello guys!

I am dealing with the same problem here. I am trying to run my app on an AWS machine. The local rendering works fine and the remote rendering returns a segmentation fault.

Sorry, @jourdain , I didn't understand your answer. Is it possible to install this vtk versions on this kind of machine or it is only possible for ParaView?

I am working with medical images and for me it would be great to use remote rendering because the data that I am working with are quite heavy. 

What would be the best approach to deal with this problem? Change the AWS machine? Install ParaView binaries? Do you have any examples?

Sorry about all the questions... but I am kind of lost on how to solve this problem 😅.

I hope you guys are well. Thank you again!

**Reply by jourdain** [reply]
This is also discussed [here](https://discourse.vtk.org/t/vtk-trame-docker-gpu-rendering/8815)

TLTR: Yes, but you need to build VTK yourself against an EGL backend or use ParaView-EGL as a VTK runtime.
